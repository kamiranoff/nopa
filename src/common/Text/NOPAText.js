import React, { PropTypes } from 'react';
import { Text } from 'react-native';

const NOPAText = ({ fontSize, fontFamily, color, textStyle, children }) => (
  <Text
    style={[{
      fontSize,
      fontFamily,
      color,
    }, textStyle ]}
  >
    {children}
  </Text>

);

NOPAText.defaultProps = {
  fontSize: 18,
  fontFamily: 'monserratSemiBold',
  color: '#444',
  textStyle: {
    marginTop: 10,
    marginBottom: 5,
  },
};

NOPAText.propTypes = {
  fontSize: PropTypes.number,
  fontFamily: PropTypes.string,
  color: PropTypes.string,
  textStyle: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.object,
  ]),
  children: PropTypes.node.isRequired,
};

export default NOPAText;
