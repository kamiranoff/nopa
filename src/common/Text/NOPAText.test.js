import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NOPAText from './NOPAText';

describe('<NOPAText />', () => {
  it('Should return the right default style', () => {
    expect(renderer.create(<NOPAText>The right text!</NOPAText>)).toMatchSnapshot();
  });

  it('Should return the right overloaded style', () => {
    expect(renderer.create(
      <NOPAText
        fontSize={18}
        fontFamily={"monserratSemiBold"}
        color={"#444"}
        onPress={() => console.log('Pressed')} // eslint-disable-line no-console
      >
        Content
      </NOPAText>)).toMatchSnapshot();
  });
});
