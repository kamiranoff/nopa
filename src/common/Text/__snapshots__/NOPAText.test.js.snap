// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NOPAText /> Should return the right default style 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#444",
        "fontFamily": "monserratSemiBold",
        "fontSize": 18,
      },
      Object {
        "marginBottom": 5,
        "marginTop": 10,
      },
    ]
  }
>
  The right text!
</Text>
`;

exports[`<NOPAText /> Should return the right overloaded style 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#444",
        "fontFamily": "monserratSemiBold",
        "fontSize": 18,
      },
      Object {
        "marginBottom": 5,
        "marginTop": 10,
      },
    ]
  }
>
  Content
</Text>
`;
